packages/web-media-box/src/dash-parser/parse.ts

```diff
@@ -1,19 +1,66 @@
-import { ParsedManifest } from "./types/parsedManifest";
+import { Attributes, ParsedManifest } from "./types/parsedManifest";
 import { testString } from "./examples/mpd";
+import { handleNode } from "./handleNode";
+import { parseAttributes } from "./parseAttributes";
 
 /**
  * Parses a MPD manifest file.
  *
- * @param playlist The URL of the mpd manifest to be parsed.
+ * @param manifest A string containing the XML of a DASH manifest.
  */
-export default function parse(playlist: string): ParsedManifest {
-  const parsedManifest: ParsedManifest = {
-    segments: [],
-    custom: {}
+export default function parse(manifest: string, url: string): ParsedManifest {
+  const initialParsedManifest: ParsedManifest = {
+    uri: url || '',
+    representations: [],
   };
 
-  // TODO: implement parsing.
-  var doc = new DOMParser().parseFromString(testString, 'text/xml');
+  const doc = new DOMParser().parseFromString(manifest, 'text/xml');
 
-  return parsedManifest;
+  return iterateXMLNodes(doc, initialParsedManifest);
+}
+
+// also going to need to keep track of BaseURL
+function iterateXMLNodes(xmlDocument: XMLDocument, manifest: ParsedManifest): ParsedManifest {
+  // TODO: Maybe use some sort of object to check state of the parser
+  // depth, bredth, current period, adaptionset, base URL, ect.
+  // const parserState: ParserState = {
+  //   node: xmlDocument,
+  //   parentNode: null,
+  //   baseURL: '',
+  //   attributes: {}
+  // }
+
+  function iterateNode(node: Node | Element, parentNode: Element | null, attributes?: any, baseUrl?: string): void {
+    // Display node name and attributes
+    // 1 is equivalent to Node.ELEMENT_NODE
+    let newAttributes = {} as Attributes
+    let response;
```

Comment:
Do you think we can avoid using a response?

ideally we would like to see something like `iterateXml((node: Node) => void)`;

We should probably avoid going through each attribute for each node.

Each NodeHandler will query attributes it needs to create a its model using `node.getAttribute()`

Reply:
We can also avoid passing parent node since we each node has a reference for its parent node, so we can query it via O(1)

Reply:
Just put out a PR that gets us closer to what I think we are going for with the iteration.